### ISSUE TITLE
Bug: `no-unused-vars` false positive on singleton pattern

### ISSUE THREAD

**Issue by HappyStriker** [eslint/eslint#19484]
### Environment

Node version: v23.9.0
npm version: 11.2.0
Local ESLint version: (global only)
Global ESLint version: v9.21.0
Operating System: macOS 15.3.1

### What parser are you using?

Default (Espree)

### What did you do?

```javascript
class Singleton {
  static {
    Singleton.instance = new Singleton();
  }

  constructor() {
    // ...
  }
}
```

### What did you expect to happen?

I would expect ESLint to properly detect the usage of `Singleton`, because there is no unused variable in this code.

### What actually happened?

ESLint shows the error `'Singleton' is defined but never used.eslint[no-unused-vars](https://eslint.org/docs/latest/rules/no-unused-vars)`.

### Link to Minimal Reproducible Example

https://eslint.org/play/#eyJ0ZXh0IjoiY2xhc3MgU2luZ2xldG9uIHtcbiAgc3RhdGljIHtcbiAgICBTaW5nbGV0b24uaW5zdGFuY2UgPSBuZXcgU2luZ2xldG9uKCk7XG4gIH1cblxuICBjb25zdHJ1Y3RvcigpIHtcbiAgICAvLyAuLi5cbiAgfVxufSIsIm9wdGlvbnMiOnsicnVsZXMiOnsiY29uc3RydWN0b3Itc3VwZXIiOlsiZXJyb3IiXSwiZm9yLWRpcmVjdGlvbiI6WyJlcnJvciJdLCJnZXR0ZXItcmV0dXJuIjpbImVycm9yIl0sIm5vLWFzeW5jLXByb21pc2UtZXhlY3V0b3IiOlsiZXJyb3IiXSwibm8tY2FzZS1kZWNsYXJhdGlvbnMiOlsiZXJyb3IiXSwibm8tY2xhc3MtYXNzaWduIjpbImVycm9yIl0sIm5vLWNvbXBhcmUtbmVnLXplcm8iOlsiZXJyb3IiXSwibm8tY29uZC1hc3NpZ24iOlsiZXJyb3IiXSwibm8tY29uc3QtYXNzaWduIjpbImVycm9yIl0sIm5vLWNvbnN0YW50LWJpbmFyeS1leHByZXNzaW9uIjpbImVycm9yIl0sIm5vLWNvbnN0YW50LWNvbmRpdGlvbiI6WyJlcnJvciJdLCJuby1jb250cm9sLXJlZ2V4IjpbImVycm9yIl0sIm5vLWRlYnVnZ2VyIjpbImVycm9yIl0sIm5vLWRlbGV0ZS12YXIiOlsiZXJyb3IiXSwibm8tZHVwZS1hcmdzIjpbImVycm9yIl0sIm5vLWR1cGUtY2xhc3MtbWVtYmVycyI6WyJlcnJvciJdLCJuby1kdXBlLWVsc2UtaWYiOlsiZXJyb3IiXSwibm8tZHVwZS1rZXlzIjpbImVycm9yIl0sIm5vLWR1cGxpY2F0ZS1jYXNlIjpbImVycm9yIl0sIm5vLWVtcHR5IjpbImVycm9yIl0sIm5vLWVtcHR5LWNoYXJhY3Rlci1jbGFzcyI6WyJlcnJvciJdLCJuby1lbXB0eS1wYXR0ZXJuIjpbImVycm9yIl0sIm5vLWVtcHR5LXN0YXRpYy1ibG9jayI6WyJlcnJvciJdLCJuby1leC1hc3NpZ24iOlsiZXJyb3IiXSwibm8tZXh0cmEtYm9vbGVhbi1jYXN0IjpbImVycm9yIl0sIm5vLWZhbGx0aHJvdWdoIjpbImVycm9yIl0sIm5vLWZ1bmMtYXNzaWduIjpbImVycm9yIl0sIm5vLWdsb2JhbC1hc3NpZ24iOlsiZXJyb3IiXSwibm8taW1wb3J0LWFzc2lnbiI6WyJlcnJvciJdLCJuby1pbnZhbGlkLXJlZ2V4cCI6WyJlcnJvciJdLCJuby1pcnJlZ3VsYXItd2hpdGVzcGFjZSI6WyJlcnJvciJdLCJuby1sb3NzLW9mLXByZWNpc2lvbiI6WyJlcnJvciJdLCJuby1taXNsZWFkaW5nLWNoYXJhY3Rlci1jbGFzcyI6WyJlcnJvciJdLCJuby1uZXctbmF0aXZlLW5vbmNvbnN0cnVjdG9yIjpbImVycm9yIl0sIm5vLW5vbm9jdGFsLWRlY2ltYWwtZXNjYXBlIjpbImVycm9yIl0sIm5vLW9iai1jYWxscyI6WyJlcnJvciJdLCJuby1vY3RhbCI6WyJlcnJvciJdLCJuby1wcm90b3R5cGUtYnVpbHRpbnMiOlsiZXJyb3IiXSwibm8tcmVkZWNsYXJlIjpbImVycm9yIl0sIm5vLXJlZ2V4LXNwYWNlcyI6WyJlcnJvciJdLCJuby1zZWxmLWFzc2lnbiI6WyJlcnJvciJdLCJuby1zZXR0ZXItcmV0dXJuIjpbImVycm9yIl0sIm5vLXNoYWRvdy1yZXN0cmljdGVkLW5hbWVzIjpbImVycm9yIl0sIm5vLXNwYXJzZS1hcnJheXMiOlsiZXJyb3IiXSwibm8tdGhpcy1iZWZvcmUtc3VwZXIiOlsiZXJyb3IiXSwibm8tdW5kZWYiOlsiZXJyb3IiXSwibm8tdW5leHBlY3RlZC1tdWx0aWxpbmUiOlsiZXJyb3IiXSwibm8tdW5yZWFjaGFibGUiOlsiZXJyb3IiXSwibm8tdW5zYWZlLWZpbmFsbHkiOlsiZXJyb3IiXSwibm8tdW5zYWZlLW5lZ2F0aW9uIjpbImVycm9yIl0sIm5vLXVuc2FmZS1vcHRpb25hbC1jaGFpbmluZyI6WyJlcnJvciJdLCJuby11bnVzZWQtbGFiZWxzIjpbImVycm9yIl0sIm5vLXVudXNlZC1wcml2YXRlLWNsYXNzLW1lbWJlcnMiOlsiZXJyb3IiXSwibm8tdW51c2VkLXZhcnMiOlsiZXJyb3IiXSwibm8tdXNlbGVzcy1iYWNrcmVmZXJlbmNlIjpbImVycm9yIl0sIm5vLXVzZWxlc3MtY2F0Y2giOlsiZXJyb3IiXSwibm8tdXNlbGVzcy1lc2NhcGUiOlsiZXJyb3IiXSwibm8td2l0aCI6WyJlcnJvciJdLCJyZXF1aXJlLXlpZWxkIjpbImVycm9yIl0sInVzZS1pc25hbiI6WyJlcnJvciJdLCJ2YWxpZC10eXBlb2YiOlsiZXJyb3IiXX0sImxhbmd1YWdlT3B0aW9ucyI6eyJwYXJzZXJPcHRpb25zIjp7ImVjbWFGZWF0dXJlcyI6e319fX19

### Participation

- [ ] I am willing to submit a pull request for this issue.

**Comment by mdjermanovic** [comment]
Hi @HappyStriker, thanks for the issue!

I'm not if this class should be considered used, but for these patterns you can use option `"ignoreClassWithStaticInitBlock": true`:

https://eslint.org/docs/latest/rules/no-unused-vars#ignoreclasswithstaticinitblock

[Playground demo](https://eslint.org/play/#eyJ0ZXh0IjoiY2xhc3MgU2luZ2xldG9uIHtcbiAgc3RhdGljIHtcbiAgICBTaW5nbGV0b24uaW5zdGFuY2UgPSBuZXcgU2luZ2xldG9uKCk7XG4gIH1cblxuICBjb25zdHJ1Y3RvcigpIHtcbiAgICAvLyAuLi5cbiAgfVxufVxuIiwib3B0aW9ucyI6eyJydWxlcyI6eyJuby11bnVzZWQtdmFycyI6WyJlcnJvciIseyJpZ25vcmVDbGFzc1dpdGhTdGF0aWNJbml0QmxvY2siOnRydWV9XX0sImxhbmd1YWdlT3B0aW9ucyI6eyJwYXJzZXJPcHRpb25zIjp7ImVjbWFGZWF0dXJlcyI6e319fX19)

**Comment by HappyStriker** [comment]
> Hi [@HappyStriker](https://github.com/HappyStriker), thanks for the issue!
> 
> I'm not if this class should be considered used, but for these patterns you can use option `"ignoreClassWithStaticInitBlock": true`:
> 
> https://eslint.org/docs/latest/rules/no-unused-vars#ignoreclasswithstaticinitblock
> 
> [Playground demo](https://eslint.org/play/#eyJ0ZXh0IjoiY2xhc3MgU2luZ2xldG9uIHtcbiAgc3RhdGljIHtcbiAgICBTaW5nbGV0b24uaW5zdGFuY2UgPSBuZXcgU2luZ2xldG9uKCk7XG4gIH1cblxuICBjb25zdHJ1Y3RvcigpIHtcbiAgICAvLyAuLi5cbiAgfVxufVxuIiwib3B0aW9ucyI6eyJydWxlcyI6eyJuby11bnVzZWQtdmFycyI6WyJlcnJvciIseyJpZ25vcmVDbGFzc1dpdGhTdGF0aWNJbml0QmxvY2siOnRydWV9XX0sImxhbmd1YWdlT3B0aW9ucyI6eyJwYXJzZXJPcHRpb25zIjp7ImVjbWFGZWF0dXJlcyI6e319fX19)

Excellent, thank you! This works like a charm.
